lib/semian/circuit_breaker.rb

```diff
@@ -30,9 +32,7 @@ def acquire
     end
 
     def request_allowed?
-      return true if closed?
-      half_open if error_timeout_expired?
-      !open?
+      closed? || half_open? || (open? && error_timeout_expired?)
```

Comment:
Indeed. But JRuby isn't really a concern at this point since we rely on C extensions.